akka-http-core/src/main/java/akka/http/javadsl/model/HttpEntity.java

```diff
@@ -92,5 +92,41 @@
      * Use getDataBytes and stream processing instead if the expected data is big or
      * is likely to take a long time.
      */
-    Future<HttpEntityStrict> toStrict(long timeoutMillis, Materializer materializer);
+    Future<HttpEntity.Strict> toStrict(long timeoutMillis, Materializer materializer);
+
+    /**
+     * The entity type which consists of a predefined fixed ByteString of data.
+     */
+    interface Strict extends UniversalEntity {
+        ByteString getData();
+    }
+
+    /**
+     * The default entity type which has a predetermined length and a stream of data bytes.
+     */
+    interface Default extends UniversalEntity {
+        long getContentLength();
+    }
+
+    /**
+     * Represents an entity without a predetermined content-length. Its length is implicitly
+     * determined by closing the underlying connection. Therefore, this entity type is only
+     * available for Http responses.
+     */
+    interface CloseDelimited extends ResponseEntity {
+    }
+
+    /**
+     * Represents an entity transferred using `Transfer-Encoding: chunked`. It consists of a
+     * stream of {@link ChunkStreamPart}.
+     */
+    interface Chunked extends RequestEntity, ResponseEntity {
+        Source<ChunkStreamPart, ?> getChunks();
```

Comment:
Please use BoxedUnit instead of `?`, type checking can be really nasty with existentials, also in Java.